### DISCUSSION TITLE
How to add addons for Project clusters

### DISCUSSION THREAD

**Discussion by clickersmudge** [kubermatic/kubermatic#11288] [Q&A] [answered]
Hi,
We created project for example "test".
In "test" project we created cluster "test1cluster".

How to add to this cluster for example "ccm-hetzner, csi-hetzner" addons.
![Zrzut ekranu z 2022-11-03 21-46-47](https://user-images.githubusercontent.com/113194014/199830671-5b4d4544-bda8-4dfd-a21a-a6b3bc6715bc.png)

We add this addon to KubeOne.
```
$ nano kubeone.yaml
apiVersion: kubeone.k8c.io/v1beta2
kind: KubeOneCluster
versions:
  kubernetes: '1.24.7'
cloudProvider:
  hetzner: {}
  external: true
# Addons are Kubernetes manifests to be deployed after provisioning the cluster
addons:
  enable: true
  # In case when the relative path is provided, the path is relative
  # to the KubeOne configuration file.
  path: "./addons"
  addons:
    - name: ccm-hetzner
    - name: cluster-autoscaler
    - name: csi-hetzner
    - name: default-storage-class
    - name: unattended-upgrades
$ kubeone apply --manifest kubeone.yaml --tfjson .
```

Addons are added to "worker node", but how to add this addon on user cluser?

**Comment by embik** [comment, marked as answer]
Hi @clickersmudge,

As mentioned [on the other thread](https://github.com/kubermatic/kubermatic/discussions/11295#discussioncomment-4073562), there is no need to install Hetzner CCM or CSI manually, this is something done by KKP automatically. KubeOne addons and KKP addons are distinct from each other. As such, the addons you installed on KubeOne do not exist in KKP. You can find more information on KKP addons [in our documentation](https://docs.kubermatic.com/kubermatic/v2.21/architecture/concept/kkp-concepts/addons/). That page includes a list of addons shipped with KKP by default and ways to configure your own.

For an alternative to addons, also check out [Applications](https://docs.kubermatic.com/kubermatic/v2.21/tutorials-howtos/applications/), a relatively recent feature addon in KKP.
